sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/SqlParser.scala

```diff
@@ -228,7 +228,12 @@ class SqlParser extends AbstractSparkSQLParser with DataTypeParser {
     andExpression * (OR ^^^ { (e1: Expression, e2: Expression) => Or(e1, e2) })
 
   protected lazy val andExpression: Parser[Expression] =
-    comparisonExpression * (AND ^^^ { (e1: Expression, e2: Expression) => And(e1, e2) })
+    booleanFactor * (AND ^^^ { (e1: Expression, e2: Expression) => And(e1, e2) })
```

Comment:
@chenghao-intel Not sure how. Adding the NOT clause in the expression rule would break precedence rules. Also, binding expression -> orExpression -> andExpression -> booleanFactor -> comparison is pretty much they way it is expressed in the grammars for standard SQL.